geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java

```diff
@@ -192,9 +194,20 @@ private void loadWebApplications() {
         logger.debug(message);
       }
     } else {
+      String[] authTokenEnabledComponents = config.getSecurityAuthTokenEnabledComponents();
+      boolean pulseOauth = Arrays.stream(authTokenEnabledComponents)
+          .anyMatch(AuthTokenEnabledComponents::hasPulse);
       String pwFile = this.config.getJmxManagerPasswordFile();
-      if (securityService.isIntegratedSecurity() || StringUtils.isNotBlank(pwFile)) {
+      if (securityService.isIntegratedSecurity()) {
+        if (pulseOauth) {
+          System.setProperty("spring.profiles.active", "pulse.authentication.oauth");
```

Comment:
this code repeats "spring.profiles.active" four times. Consider changing it to just call System.setProperty once (at the end). You could introduce a localvar "String activeProfileValue" and have each of these lines set that local var.